Change startTime_ to endTime_ in FutureInfo

xush6528 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31078 Change type of timeoutFutures_ key to time_point instead of duration
* **#30342 Change startTime_ to endTime_ in FutureInfo**

This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

xush6528 committed:
Change startTime_ to endTime_ in FutureInfo

This can eliminate the unnecessary calls to getRPCEndTime().

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

mrshenli commented on torch/csrc/distributed/rpc/process_group_agent.h:
```diff
@@ -81,16 +81,16 @@ class ProcessGroupAgent : public RpcAgent {
     FutureInfo(
         const std::shared_ptr<FutureMessage>& future,
-        const std::chrono::milliseconds& startTime,
+        const std::chrono::milliseconds& endTime_,
```
remove the trailing underscore from the var name?

mrshenli reviewed (approved):
This LGTM! Just have one minor comment on the var name.

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Thanks, this is a great enhancement.

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

pietern reviewed (changes requested):
This piggybacks unrelated changes.

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

xush6528 commented:
@pietern I re-exported. It's gone.

xush6528 committed:
Update on "Change startTime_ to endTime_ in FutureInfo"


This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
Restamp since the botched merge has been fixed now.

jjlilley commented:
fwiw, LGTM, but might need to rebase/reupload, since I believe there was a bad merge earlier. :/
